.github/workflows/image.yml

```diff
@@ -0,0 +1,75 @@
+name: image
+on:
+  pull_request:
+  push:
+    branches:
+      - master
+      - release/**
+jobs:
+  build-image:
+    strategy:
+      matrix:
+        arch: [amd64]
+    runs-on: ubuntu-latest
+    env:
+      # IMG_CACHE: ghcr.io/getsentry/symbolicator:${{ matrix.arch }}-latest
+      IMG_VERSIONED: ghcr.io/getsentry/symbolicator:${{ matrix.arch }}-${{ github.sha }}
+      NIGHTLY_IMG_CACHE: ghcr.io/getsentry/symbolicator:${{ matrix.arch }}-nightly
+      BUILDER_IMG_CACHE: ghcr.io/getsentry/symbolicator:${{ matrix.arch }}-builder
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          submodules: recursive
+      - run: docker run --rm --privileged tonistiigi/binfmt --install arm64
+        if: matrix.arch == 'arm64'
+
+      # Builder Image
+      - name: build builder img
+        run: |
+          set -euxo pipefail
+          args=()
+          if docker pull -q "$BUILDER_IMG_CACHE"; then
+            args+=(--cache-from "$BUILDER_IMG_CACHE")
+          fi
+          docker buildx build \
+              "${args[@]}" \
+              --build-arg BUILDKIT_INLINE_CACHE=1 \
+              --platform linux/${{ matrix.arch }} \
+              --tag "$BUILDER_IMG_CACHE" \
+              --target symbolicator-build \
+              .
+      - name: push builder img
+        run: |
+          set -euxo pipefail
+          docker login --username '${{ github.actor }}' --password '${{ secrets.GITHUB_TOKEN }}' ghcr.io
+          docker push "$BUILDER_IMG_CACHE"
+        if: github.event_name != 'pull_request'
```

Comment:
it should probably push everything at the end only when both builds have succeeded